app/controllers/mail_subscriptions_controller.rb

```diff
@@ -0,0 +1,41 @@
+# frozen_string_literal: true
+
+class MailSubscriptionsController < ApplicationController
+  layout 'auth'
+
+  skip_before_action :require_functional!
+
+  before_action :set_body_classes
+  before_action :set_user
+  before_action :set_type
+
+  def show; end
+
+  def create
+    @user.settings[email_type_from_param] = false
+    @user.save!
+  end
+
+  private
+
+  def set_user
+    @user = GlobalID::Locator.locate_signed(params[:token], for: 'unsubscribe')
```

Comment:
While none of these seem to be a concern, it's good to keep in mind the implications of using Signed Global IDs are:
- the following signed dict is exposed to the user: `{"gid"=>"gid://mastodon/User/1454", "purpose"=>"default", "expires_at"=>"2023-07-12T09:26:10.298Z"}`
- we can't safely rename the `User` model or change its identifier scheme
- should the `SECRET_KEY_BASE` leak, attackers will be able to completely fake any of these

I don't think any of those are actual concerns because:
- I'm not sure we already exposed the user ID anywhere (cookies are not only signed but also encrypted iirc), but I don't think doing so is an issue
- we have never needed to rename the `User` model nor change its identifier scheme, and I can't see a reason for us doing so
- since `GlobalID` is currently only used for unsubscribing to mails, the impact would be very low, and it's very likely that worse secrets would be compromised by such a leak